Week 8/RecipeApp/app/src/main/java/com/janewaitara/movieapp/ui/recipes/RecipeViewModel.kt

```diff
@@ -0,0 +1,207 @@
+package com.janewaitara.movieapp.ui.recipes
+
+import android.app.Application
+import android.net.ConnectivityManager
+import androidx.core.content.ContextCompat.getSystemService
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.viewModelScope
+import com.janewaitara.movieapp.R
+import com.janewaitara.movieapp.RecipeApplication
+import com.janewaitara.movieapp.model.Ingredient
+import com.janewaitara.movieapp.model.Recipe
+import com.janewaitara.movieapp.model.Success
+import com.janewaitara.movieapp.networking.NetworkStatusChecker
+import com.janewaitara.movieapp.repository.RoomRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class RecipeViewModel(application: Application): AndroidViewModel(application) {
```

Comment:
In the future assignment the grade criterea has asked that we don't use AndroidViewModel, and regular view model should be fine unless you're accessing the context. For this week's assignment it's fine.